x-pack/plugins/stack_connectors/server/connector_types/stack/opsgenie/connector.ts

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import crypto from 'crypto';
+import { ServiceParams, SubActionConnector } from '@kbn/actions-plugin/server';
+import { AxiosError } from 'axios';
+import { CloseAlertParamsSchema, CreateAlertParamsSchema, Response } from './schema';
+import { CloseAlertParams, Config, CreateAlertParams, Secrets } from './types';
+import * as i18n from './translations';
+
+interface ErrorSchema {
+  message?: string;
+  errors?: {
+    message?: string;
+  };
+}
+
+export class OpsgenieConnector extends SubActionConnector<Config, Secrets> {
+  constructor(params: ServiceParams<Config, Secrets>) {
+    super(params);
+
+    this.registerSubAction({
+      method: this.createAlert.name,
+      name: 'createAlert',
+      schema: CreateAlertParamsSchema,
+    });
+
+    this.registerSubAction({
+      method: this.closeAlert.name,
+      name: 'closeAlert',
+      schema: CloseAlertParamsSchema,
+    });
+  }
+
+  public getResponseErrorMessage(error: AxiosError<ErrorSchema>) {
+    return `Message: ${
+      error.response?.data.errors?.message ?? error.response?.data.message ?? i18n.UNKNOWN_ERROR
+    }.`;
```

Comment:
I think we should omit the dot. I got an error while testing and I noticed that there are two dots in the message. For example: `"Message: To perform this action, use an API key from an API integration.."`